flang/test/Lower/pointer.f90

```diff
@@ -0,0 +1,41 @@
+! RUN: bbc -emit-fir %s -o - | FileCheck %s
+
+! CHECK-LABEL: func @_QPpointertests
+subroutine pointerTests
+  ! CHECK: fir.global @_QFpointertestsEptr1 : !fir.ptr<i32>
+  integer, pointer :: ptr1 => NULL()
+  ! CHECK: %[[c0:.*]] = constant 0 : index
+  ! CHECK: [[reg1:%[0-9]+]] = fir.convert %[[c0:.*]] : (index) -> !fir.ref<none>
+  ! CHECK: [[reg2:%[0-9]+]] = fir.convert [[reg1]] : (!fir.ref<none>) -> !fir.ptr<i32>
+  ! CHECK: fir.has_value [[reg2]] : !fir.ptr<i32>
+
+  ! CHECK: fir.global @_QFpointertestsEptr2 : !fir.ptr<f32>
+  real, pointer :: ptr2 => NULL()
+  ! CHECK: %[[c0:.*]] = constant 0 : index
+  ! CHECK: [[reg1:%[0-9]+]] = fir.convert %[[c0:.*]] : (index) -> !fir.ref<none>
+  ! CHECK: [[reg2:%[0-9]+]] = fir.convert [[reg1]] : (!fir.ref<none>) -> !fir.ptr<f32>
+  ! CHECK: fir.has_value [[reg2]] : !fir.ptr<f32>
+
+  ! CHECK: fir.global @_QFpointertestsEptr3 : !fir.ptr<!fir.complex<4>>
+  complex, pointer :: ptr3 => NULL()
+  ! CHECK: %[[c0:.*]] = constant 0 : index
+  ! CHECK: [[reg1:%[0-9]+]] = fir.convert %[[c0:.*]] : (index) -> !fir.ref<none>
+  ! CHECK: [[reg2:%[0-9]+]] = fir.convert [[reg1]] : (!fir.ref<none>) -> !fir.ptr<!fir.complex<4>>
+  ! CHECK: fir.has_value [[reg2]] : !fir.ptr<!fir.complex<4>>
+
+  ! CHECK: fir.global @_QFpointertestsEptr4 : !fir.ptr<!fir.char<1>
+  character, pointer :: ptr4 => NULL()
+  ! CHECK: %[[c0:.*]] = constant 0 : index
+  ! CHECK: [[reg1:%[0-9]+]] = fir.convert %[[c0:.*]] : (index) -> !fir.ref<none>
+  ! CHECK: [[reg2:%[0-9]+]] = fir.convert [[reg1]] : (!fir.ref<none>) -> !fir.ptr<!fir.char<1>
+  ! CHECK: fir.has_value [[reg2]] : !fir.ptr<!fir.char<1>
```

Comment:
Couple of lines here look to missing the last `>` of the type.

Reply:
strange! Why didn't FileCheck catch it?
I have pushed it.
Thanks for catching.

Reply:
FileCheck is just a text compare tool, by default, it's happy if can match the whole pattern with part of one line's content, even if there is more before or after the pattern in the line.
E.g, in a test running a program:
```
  !CHECK: 10
  print * , i
```
The test will succeed even if `i` is `100`, `10.54`, `1000` or `510`, `591064.55`.

One just has to be aware of that and harden the test when needed, for instance, just above, the test should probably be `!CHECK: {{^ *}}10{{ *$}}` (or `!CHECK: 10` but running FileCheck with the `--match-full-lines` option).